### PR TITLE
Let Digest authentication act on nonce expiry

### PR DESCRIPTION
**Pull request by Frank071**
Back in 2013 marmotil discovered that libwww-perl did not correctly act on the expiry of nonce as part of Digest authentication. For some reason this was not picked up on. This pull request essentially contains the original patches made by marmotil, but adapted to the current version.

Pull requests #40, #94, #311 and #312 are obsolete after applying this.

### CODE REVIEW

**Comment by Frank071** [comment]
@oalders please come to my rescue... Travis still complains "Failed test 'Changes has content for 6.39'", but I did enter changes for that version in Changes - what am I doing wrong (and how to fix it w/o creating yet another pull request...)

File: Changes [hunk 1]
```diff
@@ -1,8 +1,6 @@
 Change history for libwww-perl
 
 {{$NEXT}}
-
-6.39      2019-03-27 19:00:00Z
```
Comment (oalders) [inline Changes]: 👍

**Comment by oalders** [comment]
Once this is all done, I think we'll want to squash these down to one commit, but let's not worry about that right now.

File: Changes (hunk 1 above)
Comment (Frank071) [reply Changes]: I checked a previous pull request and made changes accordingly. Now waiting for Travis test (but they are experiencing issues and the services is downgraded)

Comment (oalders) [reply Changes]: Good stuff.  Also, you don't need to create a new pull request even with substantial changes.  `git push --force` will overwrite your previous commits and the pull request will reflect that, which allows us to continue discussion in the same place.

Comment (Frank071) [reply Changes]: Good to know for next time - it seems it passes the test now. BTW I am not doing this via git, but via the site (don't ask ;-)) - too late I saw that I could apply changes to the comparing branch and have the pull request updated. We're getting there...

File: lib/LWP/Authen/Basic.pm [hunk 2]
```diff
@@ -11,6 +11,10 @@ sub auth_header {
     return "Basic " . MIME::Base64::encode("$user:$pass", "");
 }
 
+sub reauth_requested {
```
Comment (genio) [inline lib/LWP/Authen/Basic.pm]: Can we make these new methods private, please?

```sub _reauth_requested```

Comment (Frank071) [reply lib/LWP/Authen/Basic.pm]: Seems fine to me

Comment (oalders) [reply lib/LWP/Authen/Basic.pm]: @Frank071 are you still interested in making this change?

Comment (Frank071) [reply lib/LWP/Authen/Basic.pm]: Yes, but tell me what to do... I am not that much into Git...

**Comment by Frank071** [comment]
@oalders / @vanHoesel - I am unsure where we are now and how (I can/to) proceed... As far as I can see this pull request now blocks on a failed test for build perl-5.12-thr, which is inherited from the current master branch. It seems that ua_handlers.t simply misses two 'plan tests' lines (not sure why this is a problem for 5.12-thr only). I patched the file in my own branch, but then 5.14-thr failed on a completely different (and unrelated) test. Out of curiousity I reran that Travis test and... it succeeded. I fear my lack of github and travis knowledge is bugging me...

**Comment by oalders** [comment]
@Frank071 there's some instability in the threaded Perl tests.  I don't know if we're actually sure of why that is, so I wouldn't worry about that part of it right now.  Thanks for making these changes.

**Comment by oalders** [comment]
Am closing this as we are dealing with it in #321.  Thanks @Frank071 and @genio!

**Pull request closed without merging**